scipy/stats/tests/test_stats.py

```diff
@@ -4512,9 +4536,26 @@ def test_normalitytests():
 
 class TestRankSums(object):
     def test_ranksums_result_attributes(self):
-        res = stats.ranksums(np.arange(5), np.arange(25))
+        # this example has been taken from:
+        # http://www.sthda.com/english/wiki/unpaired-two-samples-wilcoxon-test-in-r
+        women_weight = [38.9, 61.2, 73.3, 21.8, 63.4, 64.6, 48.4, 48.8, 48.5]
+        men_weight = [67.8, 60, 63.4, 76, 89.4, 73.3, 67.3, 61.3, 62.4]
+        pvalue = 0.024194132827
+        pvalue_less, pvalue_greater = 0.0120970664135, 0.987902933586
+        res = stats.ranksums(women_weight, men_weight)
+        res_less = stats.ranksums(women_weight, men_weight,
+                                  alternative='less')
+        res_greater = stats.ranksums(women_weight, men_weight,
+                                     alternative='greater')
         attributes = ('statistic', 'pvalue')
+        assert_array_almost_equal(res.pvalue, pvalue, decimal=2)
```

Comment:
Is it not accurate to more digits? Do you know why?

Reply:
> Is it not accurate to more digits? Do you know why?

Unfortunately, they only match up to 2 digits and I don't know why.

For reference, this is the code I am using in R:

```R
options(digits=15)

women_weight <- c(38.9, 61.2, 73.3, 21.8, 63.4, 64.6, 48.4, 48.8, 48.5)
men_weight <- c(67.8, 60, 63.4, 76, 89.4, 73.3, 67.3, 61.3, 62.4)

wilcox.test(women_weight, men_weight, exact = FALSE, correct = FALSE)
```

and this is the output:

```none

	Wilcoxon rank sum test

data:  women_weight and men_weight
W = 15, p-value = 0.024194132827
alternative hypothesis: true location shift is not equal to 0
```

Am I doing something wrong?

Reply:
I don't think so. `mannwhitneyu` matches R, though.
```python3
import numpy as np
from scipy.stats import mannwhitneyu, ranksums
women_weight = [38.9, 61.2, 73.3, 21.8, 63.4, 64.6, 48.4, 48.8, 48.5]
men_weight = [67.8, 60, 63.4, 76, 89.4, 73.3, 67.3, 61.3, 62.4]
pvalue = 0.024194132827
res1 = ranksums(women_weight, men_weight)
res2 = mannwhitneyu(women_weight, men_weight, alternative='two-sided', use_continuity=False)
np.testing.assert_allclose(res2.pvalue, pvalue)
print(res1.pvalue - res2.pvalue)
```
Would you be willing to look into what mannwhitneyu is doing differently?

Reply:
> Would you be willing to look into what mannwhitneyu is doing differently?

Yeah, let me check. I will let you know if I see something interesting.

Reply:
Ok, so I found something that makes sense. We see the difference between `ranksums` and `mannwhitneyu` because `mannwhitneyu` corrects for ties while `ranksums` doesn't. In the line below:

https://github.com/scipy/scipy/blob/4fe52aed1ea375c2aa7a2a6b634787c9abdf32e3/scipy/stats/stats.py#L7417

replace `T` (here `T` is the correction for ties. see line 7414) with `1` so it becomes: `sd = np.sqrt(1 * n1 * n2 * (n1+n2+1) / 12.0)`

and then the code that you provided gives this output:

```python
>>> import numpy as np
>>> from scipy.stats import mannwhitneyu, ranksums
>>> women_weight = [38.9, 61.2, 73.3, 21.8, 63.4, 64.6, 48.4, 48.8, 48.5]
>>> men_weight = [67.8, 60, 63.4, 76, 89.4, 73.3, 67.3, 61.3, 62.4]
>>> pvalue = 0.024194132827
>>> res1 = ranksums(women_weight, men_weight)
>>> res2 = mannwhitneyu(women_weight, men_weight, alternative='two-sided', use_continuity=False)
>>> # np.testing.assert_allclose(res2.pvalue, pvalue)
>>> print(res1.pvalue - res2.pvalue)
0.0
```

I don't think there is a way to avoid correction for ties in R's `wilcox.test` so we should probably use the uncorrected value `res2.pvalue` in the code snippet above. What do you say?

Reply:
Ah, of course. That would work, but I would prefer that it's something that can be verified without changing `mannwhitneyu` code. What do you think of using `mannwhitneyu` to calculate the "ground truth" value of the statistic, then standardizing it and getting the p-value from `norm` right there in the test? Formula is pretty simple, especially without ties.

Reply:
Done. Thanks!